packages/purpose-process/src/services/validators.ts

```diff
@@ -40,3 +56,96 @@ export const isDeletableVersion = (
 
 export const isRejectable = (purposeVersion: PurposeVersion): boolean =>
   purposeVersion.state === purposeVersionState.waitingForApproval;
+
+export const assertEserviceHasSpecificMode = (
+  eservice: EService,
+  expectedMode: EServiceMode
+): void => {
+  if (eservice.mode !== expectedMode) {
+    throw eServiceModeNotAllowed(eservice.id, expectedMode);
+  }
+};
+
+export const assertConsistentFreeOfCharge = (
+  isFreeOfCharge: boolean,
+  freeOfChargeReason: string | undefined
+): void => {
+  if (isFreeOfCharge && !freeOfChargeReason) {
```

Comment:
> this check will fail and throw exception if `isFreeOfCharge == false`

It shouldn't fail since if `isFreeOfCharge == false` the condition would be not satisfited without even checking `freeOfChargeReason` and it will not throw.

In order to throw `isFreeOfCharge` has to be `true`

Reply:
my bad 😬